jobs/build-arch.Jenkinsfile

```diff
@@ -172,8 +172,9 @@ lock(resource: "build-${params.STREAM}-${basearch}") {
                 """)
             } else {
                 def yumrepos = pipecfg.source_config.yumrepos ? "--yumrepos ${pipecfg.source_config.yumrepos}" : ""
+                def variant = stream_info.variant ? "--variant ${stream_info.variant}" : ""
                 shwrap("""
-                cosa init --force --branch ${ref} --commit=${src_config_commit} ${yumrepos} ${pipecfg.source_config.url}
+                cosa init --force --branch ${ref} --commit=${src_config_commit} ${yumrepos} ${variant} ${pipecfg.source_config.url}
```

Comment:
I think we need this also at least in all the kola jobs, since it can affect denylisting. But we might as well add it in all the jobs where we do `cosa init` for consistency (e.g. I don't *think* it currently would matter for the `release` job, but it may in the future and we'd easily regress on this).

Reply:
Actually, looking at the kola harness code, I think it would do the right thing since it'd look at the variant of the build and not the currently configured variant.

I think if we do https://github.com/coreos/coreos-assembler/pull/3274/files#r1047415430, then it's easy to tell that it only matters in jobs that actually `cosa build`. This would include `bump-lockfile` too. It's not relevant for RHCOS yet, but (hopefully) will be in the future.

Reply:
Not sure I understand. You don't need to specify the variant in kola as it picks it up from the `src/config.json` file

Reply:
We run follow-up kola cloud tests in separate jobs so this initialization does not carry. But yes, indeed I misread the harness code. So then we definitely do also need this in the kola jobs too in the short-term. Updated proposal above!

But I think the right thing to do for kola there is also to read it from the build metadata (and doing https://github.com/coreos/coreos-assembler/pull/3274#discussion_r1047415430).

Reply:
https://github.com/coreos/coreos-assembler/pull/2934/files#diff-9db99bb7fefa41cf89f75a965efa109bc8349ab0a55fdb093fa46775724791cbR354 > Kola looks at `config.json` right now so we'll need to fix this for kola if we don't pass the init'ed repo to kola runs.

Reply:
Proposal in https://github.com/coreos/coreos-assembler/issues/3276